### DISCUSSION TITLE
Is it possible to disable rotation but keep zoom on the touchZoomRotate handler?

### DISCUSSION THREAD

**Discussion by tomskopek** [uber/react-map-gl#2108] [Q&A] [answered]
Basically, I want to do this: `map.touchZoomRotate.disableRotation();`

I need to disable rotation, but allow the user to still zoom in and out by double tapping and dragging. ([this functionality](https://docs.mapbox.com/mapbox-gl-js/api/handlers/#touchzoomrotatehandler:~:text=They%20can%20zoom%20with%20one%20finger%20by%20double%20tapping%20and%20dragging.%20On%20the%20second%20tap%2C%20hold%20the%20finger%20down%20and%20drag%20up%20or%20down%20to%20zoom%20in%20or%20out.))

Here is an example that replicates exactly what I need to accomplish: https://docs.mapbox.com/mapbox-gl-js/example/disable-rotation/

Currently it seems like the [touchZoomRotate prop](https://visgl.github.io/react-map-gl/docs/api-reference/map#touchzoomrotate) only accepts: `true`, `false`, or `{around: 'center'}`

Is it possible to `.disableRotation()` somehow?

I also noticed [this old/closed PR](https://github.com/visgl/react-map-gl/pull/303/files) that seemed to attempt this. Would the maintainers be open to accepting another PR adding these props if there is no other way?

**Comment by Pessimistress** [comment, marked as answer]
You can access the underlying `Map` instance by ref: https://visgl.github.io/react-map-gl/docs/api-reference/map#getmap

**Reply by tomskopek** [reply]
That worked perfectly. Thanks!

**Reply by tomskopek** [reply]
my code, for the next person who comes across this:
  
```js
const mapRef = useRef(null);
  
useEffect(() => {
  if (mapRef.current) {
    const map = mapRef.current.getMap();
    map.touchZoomRotate.disableRotation();
  }
}, [mapRef.current]);
  
return (
  <Map
    ref={mapRef}
    ...
```
